### PR TITLE
Removed the HTable name converter classes in the compat modules.

### PR DESCRIPTION
**Pull request by sagarkapare**
Removed the `HTableXXNameConverter` classes in the compat modules, since they were having the same code. Moved the implementation to the base class `HTableNameConverter`.

Build is running here - https://builds.cask.co/browse/CDAP-DUT5268-1

### CODE REVIEW

**Comment by gokulavasan** [comment]
LGTM 👍

**Comment by sagarkapare** [comment]
Running integration tests on this branch.

File: cdap-hbase-compat-1.0-cdh5.5.0/src/test/java/co/cask/cdap/data2/util/hbase/HTable10CDH550NameConverterTest.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
   public void testGetSysConfigTablePrefix() throws Exception {
     CConfiguration cConf = CConfiguration.create();
     String tablePrefix = cConf.get(Constants.Dataset.TABLE_PREFIX);
-    HTableNameConverter hBaseNameConversionUtil = new HTableNameConverterFactory().get();
+    HTableNameConverter hBaseNameConversionUtil = new HTableNameConverter();
```
Comment (gokulavasan) [inline cdap-hbase-compat-1.0-cdh5.5.0/src/test/java/co/cask/cdap/data2/util/hbase/HTable10CDH550NameConverterTest.java:29]: Do we need these separate tests?

**Comment by gokulavasan** [comment]
Question about unit tests for HTableNameConverter for each compat module. Otherwise looks good.

**Comment by sagarkapare** [comment]
Thanks @gokulavasan for the review. Will send out another PR for removing the test cases. Build is already green here - https://builds.cask.co/browse/CDAP-DUT5268-3. Squashing and merging.

**Pull request merged**